VirtoCommerce.Platform.Web/Scripts/app/security/blades/account-resetPassword.js

```diff
@@ -1,27 +1,71 @@
-﻿angular.module('platformWebApp')
-.controller('platformWebApp.accountResetPasswordController', ['$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.accounts', function ($scope, bladeNavigationService, accounts) {
+angular.module('platformWebApp')
+.controller('platformWebApp.accountResetPasswordController', ['$q', '$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.accounts', function ($q, $scope, bladeNavigationService, accounts) {
     var blade = $scope.blade;
 
     function initializeBlade() {
         blade.currentEntity = {
             newPassword: '',
-            newPassword2: ''
+            forcePasswordChange: true,
+            passwordIsValid: true,
+            minPasswordLength: 0,
+            errors: []
         };
         
         blade.isLoading = false;
     };
-    
-    $scope.saveChanges = function () {
-        if (blade.currentEntity.newPassword != blade.currentEntity.newPassword2) {
-            blade.error = 'Error: new passwords doesn\'t match!';
-            return;
-        }
 
+    $scope.validatePasswordAsync = function (value) {
+        var promise = accounts.validatePassword(JSON.stringify(value)).$promise.then(
+            function(response) {
+                blade.currentEntity.errors = [];
+
+                if (response.passwordIsValid) {
```

Comment:
This long switch/case can be simplified with a one-line loop where we can use JS `Object.keys()` to get properties names and use them to choose appropriate translation key from resources